qutebrowser/mainwindow/prompt.py

```diff
@@ -697,6 +704,9 @@ def _insert_path(self, index, *, clicked=True):
             # On Windows, when we have C:\foo and tab over .., we get C:\
             path = path.rstrip(os.sep)
 
+        if self._expands_user:
+            path = path.replace(os.path.expanduser('~'), '~', 1)
```

Comment:
There are a couple of subtleties here which still aren't tested yet:

- The behavior of `self._expands_user`, i.e. making sure that the user isn't contracted when it wasn't expanded.
- The `1` count, i.e. that a directory isn't replaced twice.

Please add those, ideally as separate test functions (and not part of the base test, one test should test one thing).